### ISSUE TITLE
Epic: Import/Export * Aren't Inverses

### ISSUE THREAD

**Issue by dgjustice** [nautobot/nautobot#254]
### Environment
* Python version:
* Nautobot version:

### Steps to Reproduce
1.
2.
3.

### Expected Behavior
Export devices, then copy/paste this into import successfully.  I haven't tested other models.

### Observed Behavior
import<>export aren't inverses.  Export spits out extra fields that import won't eat.  Import status uses slug, but export sends the Name.

```[tasklist]
- [x] #3042
- [ ] #2569
- [ ] https://github.com/nautobot/nautobot/issues/3715
- [ ] #648
- [x] #2276
- [ ] #851
- [x] #1407
- [ ] #1797
- [ ] Make Device Import/Export Inverses
```

### Related (New Feature Request)
- [ ] #1543

**Comment by glennmatthews** [comment]
Thanks for the report!

FWIW, the NetBox maintainers have previously asserted that this is "by design" - see https://github.com/netbox-community/netbox/issues/4429, https://github.com/netbox-community/netbox/issues/4656, https://github.com/netbox-community/netbox/issues/4677, https://github.com/netbox-community/netbox/issues/5270, etc.

That said, we may well want to revisit this design in Nautobot.

**Comment by jifox** [comment]
.csv header `devices`:    
  - export:  `rack_name`
  - import: `rack`

**Comment by jathanism** [comment]
We won't be able to get to this immediately, but it's worth noting that a [similar change was made in NetBox 2.11](https://github.com/netbox-community/netbox/issues/4999). We would happily welcome contributions from anyone willing to port the change over, with proper credit of course.

**Comment by jathanism** [comment]
I am of the mind that we fix this in finality in v2.x using [`django-import-export`](https://django-import-export.readthedocs.io/en/latest/) and put this issue to rest without having to maintain the problem ourselves.

**Comment by bryanculver** [comment]
Related: #648

**Comment by david-kn** [comment]
Hi, not sure if you're aware of this but I also noticed that the import requires different **data value** (_slug_ vs _name_) for different columns - at least when talking about Device import.

- See the image, which probably will be more descriptive (this is also very confusing and should be unified).
![image](https://user-images.githubusercontent.com/14314517/161283579-b5141f5f-7d84-41ae-984a-e952d3b26fc8.png)

- Also the `Required ` flag does not reflect reality. It shows that `Tenant `is not required but when missing, the import crashes with error telling me that `Tenant `actually is mandatory.

This maybe is not strictly following the import/export inverse topic but on the other hand is quite a nasty inconsistency when dealing with imports.

**Comment by martin-networker** [comment]
Adding some export/import inconsistencies which are not specifically mentioned here but are problem for us.
field "rack_number" in export has to be "rack" in import
field "rack_face" in export has to be "face" in import
field "rack_face" has value "Front" in export but in import it has to be "front"

**Comment by bryanculver** [comment]
We have issues open related to improving import/export with respect to M2Ms and usage of composite keys and natural keys but all tasks above are done and import and exports are sufficiently inverse.

**Comment by glennmatthews** [comment]
Initially implemented in #3722 and revised in #4188, #4274, and #4448.
